### PR TITLE
Add redshift monitoring per #271

### PR DESCRIPTION
**Pull request by simonrondelez**
This pr adds the chart for redshift-monitoring thru the cloudwatch-exporter

### CODE REVIEW

File: redshift-monitoring/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: "{{ .Release.Name }}-cw"
+  namespace: "{{ .Release.Namespace }}"
+  labels:
+{{ include "labels" $ | indent 4 }}
```
Comment (duboisph) [inline redshift-monitoring/templates/servicemonitor.yaml]: `{{ include "redshift-monitoring.labels" . | indent 4 }}`

File: redshift-monitoring/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {{- range $key, $val := .Values.prometheus.labels }}
```
Comment (duboisph) [inline redshift-monitoring/templates/servicemonitor.yaml]: `{{- range $key, $val := .Values.prometheusSelectorLabels }}`

File: redshift-monitoring/values.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+## Prometheus MatchLabel selectors
+prometheus:
```
Comment (duboisph) [inline redshift-monitoring/values.yaml]: Doesn't mathc your values / copy paste from different charts ;)

Replace `promethus.labels` by `prometheusSelectorLabels`

File: redshift-monitoring/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {{ $key }}: {{ $val | quote }}
+    {{- end}}
+spec:
+  endpoints:
+  - interval: {{ .Values.amazonService.interval }}
```
Comment (duboisph) [inline redshift-monitoring/templates/servicemonitor.yaml]: This value doesn't exist.

File: redshift-monitoring/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    port: http
+  namespaceSelector:
+    matchNames:
+    - "{{ .Release.Namespace }}"
+  selector:
+    matchLabels:
```
Comment (duboisph) [inline redshift-monitoring/templates/servicemonitor.yaml]: These might need updating, depending on which labels the `prometheus-cloudwatch-exporter` chart creates.

Comment (duboisph) [reply redshift-monitoring/templates/servicemonitor.yaml]: Ideally it already follows the new convention

```
      app.kubernetes.io/name: prometheus-cloudwatch-exporter
      app.kubernetes.io/instance: {{ .Release.Name }}
```

**Review by duboisph — CHANGES_REQUESTED** [review]
Missing requirements for the `prometheus-cloudwatch-exporter` sub chart.

**Review by duboisph — APPROVED** [review]

**Review by minniux — APPROVED** [review]

**Pull request merged**